library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -1275,6 +1301,22 @@ public boolean changeLocationUpdates(long updateInterval, long fastestUpdateInte
         return false;
     }
 
+    @Override
+    public boolean isKujakuLayerAdded(@NonNull KujakuLayer kujakuLayer) {
+        String[] layerIds = kujakuLayer.getLayerIds();
+        if (mapboxMap != null && mapboxMap.getStyle() != null && mapboxMap.getStyle().isFullyLoaded()) {
+            for (String layerId: layerIds) {
+                if (mapboxMap.getStyle().getLayer(layerId) == null) {
+                    return false;
+                }
+            }
+
+            return true;
```

Comment:
Can this be moved out of the conditional to be the default return value?

Reply:
From the code on the lines below this, the default return is `false`. Do we have a reason to change this to `true` instead of `false` from the conditionals

Reply:
Returning false by default is OK.

However, I'm not sure if the method should return false if `!mapboxMap.getStyle().isFullyLoaded()`. The layer could still have been added, no?

Reply:
The method is checking to see if the layers that make up the `KujakuLayer` are added. If not but the style is loaded, then the layers aren't added

Reply:
I understand, but what if the style is not fully loaded but the KujakuLayer is already added in the partially loaded style?

Reply:
A layer cannot be added when the style is partially loaded. Mapbox does not recognise such a state 
https://docs.mapbox.com/help/tutorials/first-steps-android-sdk/#configure-your-mapview

```
mapView.getMapAsync(new OnMapReadyCallback() {
@Override
public void onMapReady(@NonNull MapboxMap mapboxMap) {
mapboxMap.setStyle(Style.MAPBOX_STREETS, new Style.OnStyleLoaded() {
      @Override
      public void onStyleLoaded(@NonNull Style style) {
        // Map is set up and the style has loaded. Now you can add data or make other map adjustments
      }
    });
    }
});
```

Reply:
OK.